src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -302,6 +340,10 @@ private PostContingencyResult runPostContingencySimulation(LfNetwork network, Ac
                 network.getBranches().stream().filter(b -> !lfContingency.getBranches().contains(b)),
                 network.getBuses().stream().filter(b -> !lfContingency.getBuses().contains(b)),
                 postContingencyLimitViolations);
+            StateMonitor stateMonitor = monitors.get(lfContingency.getContingency().getId());
+            if (stateMonitor != null && stateMonitor.getContingencyContext().getContextType() == ContingencyContextType.SPECIFIC) {
```

Comment:
Maybe you have to implement the `ALL` mode in a different way to be more efficient, but I am not sure about that.